cogs/tmpc.py

```diff
@@ -434,24 +472,80 @@ async def nomod(self, ctx: Context):
                              description=f"Mods can't see or join this channel anymore")
         await ctx.reply(embed=embed)
 
-    async def check_tmpc_channel(self, ctx: Context, is_mod: bool = False, everyone: bool = False) \
+    @tmpc.command(pass_context=True,
+                  brief="A user to your temp channel",
+                  help="Default add user to the temp channel, for gaming channel the first argument must be 'False'!\n"
+                       "Member must be invited by mentions.\n"
+                       "Limitations:\n"
+                       "- This command can be used 3/hour/user\n"
+                       "- The invite command is limited to 10 user at once")
+    @cooldown(3, 3600, BucketType.user)
+    async def invite(self, ctx: Context, study_channel: bool):
+
+        if study_channel:
+            channel = await self.study_db.find_one({DBKeyWrapperEnum.OWNER.value: ctx.author.id})
+        else:
+            channel = await self.gaming_db.find_one({DBKeyWrapperEnum.OWNER.value: ctx.author.id})
+        channel: Union[GamingChannel, StudyChannel]
+        document = await self.check_tmpc_channel(ctx.author, channel.chat.id)
+
+        overwrites = document.voice.overwrites
+        msg: Message = ctx.message
+        if len(msg.mentions) > 10:
+            await ctx.reply("Warning hit limit")
+
+        changed = False
+        for mention in msg.mentions[:10]:
+            if mention not in overwrites:
+                changed = True
+                overwrites[mention] = PermissionOverwrite(connect=True,
+                                                          view_channel=True)
+        if changed:
+            await document.voice.edit(overwrites=overwrites)
+            await document.chat.edit(overwrites=overwrites)
+
+    @tmpc.command(pass_context=True,
+                  brief="Removes personal rights.",
+                  help="Limitations:"
+                       "- This command can be used 4/hour/user\n")
+    @cooldown(1, 3600, BucketType.channel)
+    async def leave(self, ctx: Context):
+        document = await self.check_tmpc_channel(ctx.author, ctx.channel.id, everyone=True)
+        if ctx.author.id == document.owner.id:
+            return
+        overwrites: dict = document.voice.overwrites
+        changed = False
+        if moderator.item in ctx.author.roles:
+            try:
+                overwrites.pop(ctx.author)
+                changed = True
+            except KeyError:
+                pass
+        else:
+            changed = True
+            overwrites[ctx.author] = PermissionOverwrite(connect=False,
+                                                         view_channel=False)
+        if changed:
+            await document.voice.edit(overwrites=overwrites)
+            await document.chat.edit(overwrites=overwrites)
+
+    async def check_tmpc_channel(self, member: Member, _id: int, is_mod: bool = False, everyone: bool = False) \
```

Comment:
Why do we need the document id all of the sudden? the reason for this method is to find the document in the first place

Reply:
That's not correct we used `ctx` before to gain the temp channel text chat id and the author.
But the invite command needs to be written in a global visible bot command chat. In order to use this method for invite too I changed the signature to give the same information as before but independent of `ctx`

Reply:
Ok but then you still have code where the document is searched first and then this Method is called. Like in Line 499 for example

Reply:
Yea but that's the only one time we need that to consider `StudyChannel` or `GamingChannel` for the command execution.
Anywhere else we get the info by the `ctx` variable because all other commands must be used in that channel.

We also could perform the every-time before we call `check_tmpc_channel` and give `that method `member` and `document` instead.

Reply:
Ah ok yeah then nevermind I forgot that the command needs to differentiate between Study and Gaming channel